### PR TITLE
Use KrillVersion for keystore based on the Cargo.toml version. (#521)

### PR DESCRIPTION
**Pull request by timbru**

### CODE REVIEW

File: src/commons/util/mod.rs [hunk 1]
```diff
@@ -1,16 +1,230 @@
...
+//------------ KrillVersion --------------------------------------------------
+
+/// Defines a Krill version. Will 
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct KrillVersion {
+    major: u64,
+    minor: u64,
+    patch: u64,
+    release: KrillVersionRelease
```
Comment (ximon18) [inline src/commons/util/mod.rs:31]: s/KrillVersionRelease/ReleaseType/ ? (version threw me off as we already express the version here in terms of major, minor, patch).

File: src/commons/util/mod.rs [hunk 2]
```diff
@@ -1,16 +1,230 @@
...
+                KrillVersionRelease::Release => Ordering::Equal,
+                _ => Ordering::Greater
+            },
+            KrillVersionRelease::Candidate(nr) => match other {
+                KrillVersionRelease::Release => Ordering::Less,
+                KrillVersionRelease::Candidate(nr_other) => nr.cmp(nr_other),
+                &KrillVersionRelease::Dev(_) => Ordering::Greater
+            },
+            KrillVersionRelease::Dev(text) => match other {
+                KrillVersionRelease::Dev(text_other) => text.cmp(text_other),
```
Comment (ximon18) [inline src/commons/util/mod.rs:203]: Maybe dev should always equal dev?

**Review by ximon18 — APPROVED** [review]

File: src/commons/util/mod.rs (hunk 2 above)
Comment (timbru) [reply src/commons/util/mod.rs]: yes, changed now

File: src/commons/util/mod.rs (hunk 1 above)
Comment (timbru) [reply src/commons/util/mod.rs]: Renamed to KrillVersionReleaseType. A bit wordy but okay and descriptive methinks.
